scipy/signal/signaltools.py

```diff
@@ -463,6 +460,9 @@ def convolve(in1, in2, mode='full'):
     volume = asarray(in1)
     kernel = asarray(in2)
 
+    if not _is_valid_mode_shapes(mode, volume.shape, kernel.shape):
+        volume, kernel = kernel, volume
```

Comment:
The reason for swapping is I think some of the code is still written to expect the "larger" array first (e.g. `fftconvolve`), so the swapping helps to resolve that issue across the board, even if it may not be necessary.